File: java/chainofresponsibility/SecurityChecksMiddlewareHandler.java
package chainofresponsibility;

public class SecurityChecksMiddlewareHandler extends AbstractMiddlewareHandler {

    @Override
    public boolean handle(Request request){
        System.out.println("Checking if request pass security checks...");
        if (!request.isHasPassedSecurityChecks()) {
            System.out.println("request failed to pass security checks...");
            return false;
        }
        return super.handle(request);
    }
}

File: typescript/chainofresponsibility/securityChecks-middleware-handler.ts
import {AbstractMiddlewareHandler} from "./abstractMiddleware-handler";
import {ChainRequest} from "./request";

export class SecurityChecksMiddlewareHandler extends AbstractMiddlewareHandler {
  public handle(request: ChainRequest): boolean {
    if (!request.isHasPassedSecurityChecks) {
      console.log("SecurityChecksMiddlewareHandler: User is not secured.");
      return false;
    }
    console.log("SecurityChecksMiddlewareHandler: User is secured.");
    return super.handle(request);
  }
}
